src/util/util_small_vector.h

```diff
@@ -33,13 +33,37 @@ namespace dxvk {
     };
 
     small_vector(small_vector&& other) {
-      if (other.m_capacity == N) {
+      if (other.m_size < N) {
         for (size_t i = 0; i < other.m_size; i++) {
-          u.m_data[i] = std::move(other.u.m_data[i]);
+          new (&u.m_data[i]) T(std::move(*other.ptr(i)));
         }
       } else {
         u.m_ptr = other.u.m_ptr;
-        other.m_capacity = N;
+        m_capacity = other.m_capacity;
+        other.u.m_ptr = nullptr;
+        other.m_capacity = 1;
```

Comment:
why 1? We initialize this to N for obvious reasons, it shouldn't ever be smaller.

Reply:
I just wanted anything that's smaller than N of the other vendor to avoid calling free on the pointer after it has been moved to the new vector.